File: testes/algoritmos-ordenacao/python/bubble-sort.py
def bubble_sort(arr):
    n = len(arr)

    for i in range(n - 1):
        for j in range(n - 1):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]

def executar_bubble_sort(numero_repeticoes, vetor, tipo_instancia):
    soma_tempos = 0
    soma_consumos_memoria = 0
    media_tempos = 0
    media_consumos_memoria = 0
    tempos_bubble_sort = []
    consumos_memoria_bubble_sort = []

    for i in range(numero_repeticoes):
        tempo_inicial = 0
        tempo_final = 0
        memoria_inicial = 0
        memoria_final = 0

        memoria_inicial = sys.getsizeof(vetor)
        tempo_inicial = time.time()
        bubble_sort(vetor.copy())
        tempo_final = time.time()
        memoria_final = sys.getsizeof(vetor)

        tempos_bubble_sort.append((tempo_final - tempo_inicial) * 1000)  # Em milissegundos
        consumos_memoria_bubble_sort.append(memoria_final - memoria_inicial)

    # Cálculo da média de tempo gasto
    soma_tempos = sum(tempos_bubble_sort)
    media_tempos = soma_tempos / numero_repeticoes

    # Cálculo da média de consumo de memória
    soma_consumos_memoria = sum(consumos_memoria_bubble_sort)
    media_consumos_memoria = soma_consumos_memoria / numero_repeticoes

    # Log de saída
    print(f"""
        Algoritmo: Bubble Sort
        Tipo da Instância/Tamanho: {tipo_instancia}/{len(vetor)}
        Qtd. de Iterações: {numero_repeticoes}
        Média de Tempo Gasto: {media_tempos:.6f} ms
        Média de Consumo de Memória: {media_consumos_memoria:.4f} bytes
    """)

File: testes/algoritmos-ordenacao/js/bubble-sort.js
export function bubbleSort(vetor) {
  let tamanhoVetor = vetor.length;

  for (let i = 0; i < (tamanhoVetor-1); i++) {
    for (let j = 0; j < (tamanhoVetor-1); j++) {
      if (vetor[j] > vetor[j + 1]) {
        let auxiliar = vetor[j];
        vetor[j] = vetor[j + 1];
        vetor[j + 1] = auxiliar;
      }
    }
  }
}

export function executarBubbleSort(numeroRepeticoes, vetor, tipo_instancia) {
  let somaTempos = 0;
  let somaConsumosMemoria = 0;
  let mediaTempos = 0;
  let mediaConsumosMemoria = 0;
  let temposBubbleSort = [];
  let consumosMemoriaBubbleSort = [];

  for (let i = 0; i < numeroRepeticoes; i++) {
    let tempoInicial = 0;
    let tempoFinal = 0;
    let memoriaInicial = 0;
    let memoriaFinal = 0;

    memoriaInicial = process.memoryUsage().heapUsed;
    tempoInicial = performance.now().toFixed(6);
    bubbleSort(vetor);
    tempoFinal = performance.now().toFixed(6);
    memoriaFinal = process.memoryUsage().heapUsed;

    temposBubbleSort.push(tempoFinal - tempoInicial);
    consumosMemoriaBubbleSort.push(memoriaFinal - memoriaInicial);
  }

  //Cálculo da média de tempo gasto
  for (let i = 0; i < temposBubbleSort.length; i++) {
    somaTempos += temposBubbleSort[i];
  }
  mediaTempos = somaTempos / temposBubbleSort.length;

  //Cálculo da média de consumo de memória
  for (let i = 0; i < consumosMemoriaBubbleSort.length; i++) {
    somaConsumosMemoria += consumosMemoriaBubbleSort[i];
  }
  mediaConsumosMemoria = somaConsumosMemoria / consumosMemoriaBubbleSort.length;

  //Log de saída
  console.log(`
        Algoritmo: Bubble Sort
        Tipo da Instância/Tamanho: ${tipo_instancia}/${vetor.length}
        Qtd. de Iterações: ${numeroRepeticoes}
        Média de Tempo Gasto: ${mediaTempos.toFixed(6)} ms
        Média de Consumo de Memória: ${mediaConsumosMemoria.toFixed(4)} bytes
    `);
}
